### PR TITLE
PXB-1831: xbcloud - add new storage types

### PR DESCRIPTION
**Pull request by borys-belinsky**

### CODE REVIEW

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst [hunk 1]
```diff
@@ -28,40 +35,237 @@ Version specific information
...
   :option:`xbcloud --swift-auth-url`
 
-Usage
---------------------------------------------------------------------------------
+Supported Cloud Storage Types
+================================================================================
+
+In addition to Swift, which has been the only option for storing backups in a
+cloud storage until |Percona XtraBackup| 2.4.14, |xbcloud| supports *Amazon
+S3*, *MinIO*, and *Google Cloud Storage*.
```
Comment (gl-sergei) [inline storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: other S3-compatible storage providers should also work (like Wasabi, Digital Ocean spaces, etc...)

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst [hunk 2]
```diff
@@ -28,40 +35,237 @@ Version specific information
...
+     - Use supply the AWS bucket name
+   * - --s3-region
+     - Use to specify the AWS region. The default value is **us-east-1**
+   * - --s3-api-version = <AUTO|2|4>
+     - Select the signing algorithm. The default value is AUTO. In this case, |xbcloud| will probe.
+   * - --s3-bucket-lookup = <AUTO|PATH|DNS>
+     - Specify whether to use **bucket.endpoint.com** or *endpoint.com/bucket**
+       style requests. The default value is AUTO. In this case, |xbcloud| will probe.
+
+The following environment variables are recognized:
```
Comment (gl-sergei) [inline storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: leftover?

**Review by gl-sergei — CHANGES_REQUESTED** [review]
Please also mention new --md5 and --header options and usage examples (see description and examples in the issue comments)

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst [hunk 3]
```diff
@@ -28,40 +35,237 @@ Version specific information
...
+
+The following options are available when using |gcs|:
+
+.. hlist::
+   :columns: 2
+
+   - --google-access-key = <ACCESS KEY ID>
+   - --google-secret-key = <SECRET ACCESS KEY>
+   - --google-bucket = <BUCKET NAME>
+   - --google-region = <REGION> (default is us-central-1)
```
Comment (gl-sergei) [inline storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: This option has no effect and will be removed. Lets not include it into the doc

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst (hunk 1 above)
Comment (dbadapt) [reply storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: We can just say something like,  "other S3 compatible storage implementations may also work but are not officially supported at this time"

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst (hunk 3 above)
Comment (dbadapt) [reply storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: https://jira.percona.com/browse/PXB-1848

**Review by gl-sergei — APPROVED** [review]

**Pull request merged**